source/fundamentals/crud/write-operations/change-a-document.txt

```diff
@@ -0,0 +1,218 @@
+   
+Return Values
+~~~~~~~~~~~~~
+
+The ``UpdateOne()`` and ``UpdateMany()`` methods return an `UpdateResult <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#UpdateResult>`__ type that
+contains information about the update operation if the operation is
+successful. The ``UpdateResult`` type contains the following properties:
+
+- ``MatchedCount`` - the number of documents matched by the filter
+- ``ModifiedCount`` - the number of documents modified by the operation
+- ``UpsertedCount`` - the number of documents upserted by the operation
+- ``UpsertedID`` - the ``_id`` field of the upserted document, or nil if
+  no upsert was performed
+
+If multiple documents match the query filter passed to ``UpdateOne()``,
+the method selects and updates the first matched document. Your update
+operation will fail if no documents match the query filter. See our **<TODO:
+link>** upsert guide to learn how to insert a new document if no
+documents match the query filter.
+
+Example
+~~~~~~~
+
+The following document describes an employee:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Analyst",
+      "bonus" : 2500,
+      ...
+   }
+
+The following example uses the ``UpdateOne()`` method to:
+
+- Match the document where the ``_id`` value is 2158.
+- Set the ``name`` field to "Mary Wollstonecraft Shelley" and the
+  ``role`` field to "Marketing Director".
+- Increment the value of the ``bonus`` field by 2000.
+
+.. code-block:: go
+
+   filter := bson.D{{"_id", 2158}}
+   update := bson.D{{"$set", bson.D{{"name", "Mary Wollstonecraft Shelley"}, 
+       {"role", "Marketing Director"}}, {"$inc", bson.D{{"bonus", 2000}}}}
+
+   result, err := collection.UpdateOne(context.TODO(), filter, update)
+   fmt.Printf("Documents matched: %v\n", result.MatchedCount)
+   fmt.Printf("Documents updated: %v\n", result.ModifiedCount)
+
+The expected output of the preceding example is as follows:
+
+.. code-block:: none
+   :copyable: false
+
+   Documents matched: 1
+   Documents updated: 1
+
+The following shows the updated document resulting from the preceding update operation:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+      "_id" : 2158,
+      "name" : "Mary Wollstonecraft Shelley",
+      "department" : "Marketing",
+      "role" : "Marketing Director",
+      "bonus" : 4500,
+      ...
+   }
+
+.. _replacementDocument:
+
+Replace
+-------
+
+To perform a replace operation, use the `ReplaceOne() <https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#Collection.ReplaceOne>`__ method and
+specify a **replacement document**, which is the document that you want
+to take the place of an existing document. Replacement documents use the
+following format:
+
+.. code-block:: go
+
+   bson.D{{"<field>", "<value>"}, {"<field>", "<value>"}, ... }
+
+Return Values
+~~~~~~~~~~~~~
+
+``ReplaceOne()`` returns an ``UpdateResult`` type that
```

Comment:
Suggestion: I think you should link to the API docs for Update result here since it's the first mention in this section